[rocprofiler-systems] Consolidate CTests to tests/ folder

kcossett-amd opened the pull request:
## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->
In preparation for the`ctest` build, it is useful to centralize all current CTests to the tests/ folder. 

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->
Moved test code under `source/bin/tests` to `tests/rocprof-sys-binary-tests.cmake`
Moved test code under `source/python/CMakeLists.txt` to `tests/rocprof-sys-python-console-tests.cmake`
Regex located in `source/bin/tests` was moved to `tests/rocprof-sys-testing.cmake`
Function `ROCPROFILER_SYSTEMS_ADD_BIN_TEST` moved to `tests/rocprof-sys-testing.cmake`

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->
Workflows
MI300
## Test Result

<!-- Briefly summarize test outcomes. -->
Workflows - **All workflows pass** (Excluding 3 TheRock CI and Azure)
  - About the Azure failures: CTests that failed are ones that fail on other PR's (that is, all the moved CTests pass).

MI300 - All **moved** tests pass
## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

kcossett-amd committed:
Consolidate CTests to tests/ folder

kcossett-amd committed:
Consolidate CTests to tests/ folder

kcossett-amd committed:
Remove comment

kcossett-amd committed:
Remove comment

adjordje-amd commented:
Maybe we can move `source/thread_limit.cpp` to the examples directory. Also, the same is applicable for the sleeper, even though is not doing anything, it is still example used to verify some rocprof-sys functionality.

kcossett-amd committed:
Fix branch
Please enter a commit message to explain why this merge is necessary,

kcossett-amd commented:
@adjordje-amd Any suggestions on how I should move `sleeper.cpp`? As it is only used in one CTest, should I move it under `examples/` and put it under a new folder called `generic`? 

Edit: I have removed sleeper.cpp and replaced it with linux sleep cmd.

kcossett-amd committed:
Separate source code and test code for thread-limit into appropriate folders

kcossett-amd committed:
Remove sleeper.cpp and instead use linux sleep cmd

kcossett-amd committed:
Merge branch 'develop' into users/kcossett-amd/rocprof-sys-structure-change

dgaliffiAMD commented on projects/rocprofiler-systems/tests/rocprof-sys-python-console-tests.cmake:
It would be okay to move this into the `rocprof-sys-python-tests.cmake` too, I think

kcossett-amd committed:
Merge remote-tracking branch 'origin/develop' into users/kcossett-amd/rocprof-sys-structure-change

kcossett-amd committed:
Merge python-console tests into python-tests

dgaliffiAMD commented:
/AzurePipelines run rocm-ci-caller

kcossett-amd committed:
Merge branch 'develop' into users/kcossett-amd/rocprof-sys-structure-change
